File: frontend/erpclink-web/src/app/features/queue/models/queue.models.ts
export type QueueStatus = 'Waiting' | 'Called' | 'InService' | 'Completed' | 'Skipped' | 'Cancelled';

export type QueuePriority = 'Normal' | 'Urgent';

export interface CheckInRequest {
  appointmentId: string;
  priority: string | null;
}

export interface TodayQueueParams {
  clinicId?: string;
  doctorId?: string;
  status?: string;
  priority?: string;
}

export interface QueueListItem {
  id: string;
  queueNumber: string;
  patientId: string;
  appointmentId: string;
  doctorId: string;
  clinicId: string;
  queueDate: string;
  priority: QueuePriority;
  status: QueueStatus;
  checkInTimeUtc: string;
  calledTimeUtc: string | null;
  serviceStartTimeUtc: string | null;
  serviceEndTimeUtc: string | null;
}

export interface QueueEntry extends QueueListItem {
  organizationId: string;
  branchId: string;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
}

File: src/Modules/Queue/ErpClink.Modules.Queue.Application/Entries/Models/QueueModels.cs
namespace ErpClink.Modules.Queue.Application.Entries.Models;

public sealed record CheckInRequest(Guid AppointmentId, string? Priority);

public sealed record SearchQueueRequest(
    DateOnly? Date,
    DateOnly? FromDate,
    DateOnly? ToDate,
    Guid? PatientId,
    Guid? AppointmentId,
    Guid? DoctorId,
    Guid? ClinicId,
    string? Status,
    string? Priority,
    string? QueueNumber,
    string? SortBy,
    bool SortDescending = false,
    int Page = 1,
    int PageSize = 20);

public sealed record TodayQueueRequest(
    Guid? ClinicId,
    Guid? DoctorId,
    string? Status,
    string? Priority);

public sealed record QueueEntryDto(
    Guid Id,
    Guid OrganizationId,
    Guid BranchId,
    string QueueNumber,
    Guid PatientId,
    Guid AppointmentId,
    Guid DoctorId,
    Guid ClinicId,
    DateOnly QueueDate,
    string Priority,
    string Status,
    DateTime CheckInTimeUtc,
    DateTime? CalledTimeUtc,
    DateTime? ServiceStartTimeUtc,
    DateTime? ServiceEndTimeUtc,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy);

public sealed record QueueListItemDto(
    Guid Id,
    string QueueNumber,
    Guid PatientId,
    Guid AppointmentId,
    Guid DoctorId,
    Guid ClinicId,
    DateOnly QueueDate,
    string Priority,
    string Status,
    DateTime CheckInTimeUtc,
    DateTime? CalledTimeUtc,
    DateTime? ServiceStartTimeUtc,
    DateTime? ServiceEndTimeUtc);

public sealed record PagedQueueResult(
    IReadOnlyList<QueueListItemDto> Items,
    int Page,
    int PageSize,
    int TotalCount);
